### PR TITLE
Downgrade to gradle 7.3.1

### PR DESCRIPTION
**Pull request by mvanbeusekom**
Downgrade the Android Gradle plugin from 8.0.2 to 7.3.1, so the project is inline with the current stable version of Flutter.

Reason for downgrading were some issues in the example app using the url_launcher plugin.

## Pre-launch Checklist

- [x] I made sure the project builds.
- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is does not need version changes.
- [x] I updated `CHANGELOG.md` to add a description of the change.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I rebased onto `main`.
- [x] I added new tests to check the change I am making, or this PR does not need tests.
- [x] I made sure all existing and new tests are passing.
- [x] I ran `dart format .` and committed any changes.
- [x] I ran `flutter analyze` and fixed any errors.

[Contributor Guide]: https://github.com/Baseflow/flutter-geocoding/blob/master/CONTRIBUTING.md
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning

### CODE REVIEW

File: geocoding_android/example/android/build.gradle [hunk 1]
```diff
@@ -26,4 +28,4 @@ subprojects {
 
 tasks.register("clean", Delete) {
     delete rootProject.buildDir
-}
+}
```
Comment (JeroenWeener) [inline geocoding_android/example/android/build.gradle:31]: Nit: add newline

File: geocoding_android/example/pubspec.yaml [hunk 2]
```diff
@@ -25,19 +25,13 @@ dependencies:
   # The following adds the Cupertino Icons font to your application.
   # Use with the CupertinoIcons class for iOS style icons.
   cupertino_icons: ^1.0.2
-  url_launcher: ^6.0.2
+  url_launcher: ^6.0.36
```
Comment (JeroenWeener) [inline geocoding_android/example/pubspec.yaml:28]: `url_launcher: 6.0.36` does not exist. Is this a typo?

**Review by JeroenWeener — CHANGES_REQUESTED** [review]
The `url_launcher` version seems incorrect due to a typo.

FYI: build is displaying `e: /home/runner/.gradle/caches/modules-2/files-2.1/androidx.annotation/annotation-jvm/1.6.0/a7257339a052df0f91433cf9651231bbb802b502/annotation-jvm-1.6.0.jar!/META-INF/annotation.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.8.0, expected version is 1.6.0.`. I am not sure if this is a problem.

File: geocoding_android/example/pubspec.yaml (hunk 2 above)
Comment (mvanbeusekom) [reply geocoding_android/example/pubspec.yaml]: That was indeed a mistake, I mixed the `url_launcher` and `url_launcher_android` packages.

**Review by JeroenWeener — APPROVED** [review]

**Pull request merged**